### PR TITLE
api: deprecate PutObject() resuming capabilities.

### PR DESCRIPTION
**Pull request by harshavardhana**
This is done to ensure our functionality to be
similar to AWS SDKs behavior. Since resuming
as a functionality itself is not expected or
supported by AWS S3.

This is an ongoing simplification with
the new changes coming in to our minio server
gateway and fs backend.

### CODE REVIEW

**Comment by vadmeste** [comment]
@harshavardhana, there is some conflicts, can you rebase again ?

**Comment by harshavardhana** [comment]
> @harshavardhana, there are some conflicts, can you rebase again ?

Done @vadmeste

**Comment by harshavardhana** [comment]
@vadmeste can you take a look again? - let me know if any changes are needed..

File: api.go [hunk 1]
```diff
@@ -575,8 +608,17 @@ func (c Client) newRequest(method string, metadata requestMetadata) (req *http.R
 		return nil, err
 	}
 
+	// Go net/http notoriously closes the request body.
+	// - The request Body, if non-nil, will be closed by the underlying Transport, even on errors.
```
Comment (vadmeste) [inline api.go]: If golang's Transport module closes contentBody, why do we close it in executeMethod() ?

Comment (harshavardhana) [inline api.go]: We shouldn't let it close it.. only once we have exhausted retries.

Comment (vadmeste) [inline api.go]: > We shouldn't let it close it.. only once we have exhausted retries.

yes, right, now I understand.

**Pull request merged**